lib/fog/google/models/storage_json/file.rb

```diff
@@ -0,0 +1,146 @@
+require "fog/core/model"
+
+module Fog
+  module Google
+    class StorageJSON
+      class File < Fog::Model
+        identity :key, :aliases => "Key"
+
+        # TODO: Verify
+        attribute :cache_control,       :aliases => "Cache-Control"
+        attribute :content_disposition, :aliases => "Content-Disposition"
+        attribute :content_encoding,    :aliases => "Content-Encoding"
+        attribute :content_length,      :aliases => ["Content-Length", "Size"], :type => :integer
+        attribute :content_md5,         :aliases => "Content-MD5"
+        attribute :content_type,        :aliases => "Content-Type"
+        attribute :etag,                :aliases => %w(Etag ETag)
+        attribute :expires,             :aliases => "Expires"
+        attribute :last_modified,       :aliases => ["Last-Modified", "LastModified"]
+        attribute :metadata
+        attribute :owner,               :aliases => "Owner"
+        attribute :storage_class,       :aliases => ["x-goog-storage-class", "StorageClass"]
+
+        # TODO: Verify
+        def acl=(new_acl)
+          valid_acls = ["private", "public-read", "public-read-write", "authenticated-read"]
+          unless valid_acls.include?(new_acl)
+            raise ArgumentError.new("acl must be one of [#{valid_acls.join(', ')}]")
+          end
+          @acl = new_acl
+        end
+
+        # TODO: Verify
+        def body
+          attributes[:body] ||= last_modified && (file = collection.get(identity)) ? file.body : ""
+        end
+
+        # TODO: Verify
+        def body=(new_body)
+          attributes[:body] = new_body
+        end
+
+        attr_reader :directory
+
+        # TODO: Verify
+        def copy(target_directory_key, target_file_key)
+          requires :directory, :key
+          service.copy_object(directory.key, key, target_directory_key, target_file_key)
+          target_directory = service.directories.new(:key => target_directory_key)
+          target_directory.files.get(target_file_key)
+        end
+
+        # TODO: Verify
+        def destroy
+          requires :directory, :key
+          begin
+            service.delete_object(directory.key, key)
+          rescue Excon::Errors::NotFound
+          end
+          true
+        end
+
+        # TODO: Verify
+        remove_method :metadata
+        def metadata
+          attributes.reject { |key, _value| !(key.to_s =~ /^x-goog-meta-/) }
+        end
+
+        # TODO: Verify
+        remove_method :metadata=
+        def metadata=(new_metadata)
+          merge_attributes(new_metadata)
+        end
+
+        # TODO: Verify
+        remove_method :owner=
```

Comment:
Got it.

2015-12-07 12:01 GMT-02:00 Nat Welch <notifications@github.com>:

> In lib/fog/google/models/storage_json/file.rb
> <https://github.com/fog/fog-google/pull/96#discussion_r46822952>:
>
> > +        end
> > +
> > +        # TODO: Verify
> > +        remove_method :metadata
> > +        def metadata
> > +          attributes.reject { |key, _value| !(key.to_s =~ /^x-goog-meta-/) }
> > +        end
> > +
> > +        # TODO: Verify
> > +        remove_method :metadata=
> > +        def metadata=(new_metadata)
> > +          merge_attributes(new_metadata)
> > +        end
> > +
> > +        # TODO: Verify
> > +        remove_method :owner=
>
> Direct copy from the storage folder. The storage code is from before my
> time, I think I'm gonna have to nuke most of this, but I want to keep the
> API the same except for Auth.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fog/fog-google/pull/96/files#r46822952>.
>



-- 
-----------------------------------------------------
Paulo Henrique Lopes Ribeiro
Senior Ruby Engineer